community/kernel/src/test/java/org/neo4j/kernel/api/index/util/FailureStorageTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.api.index.util;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.File;
+
+import org.neo4j.test.EphemeralFileSystemRule;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import static java.lang.String.format;
+
+public class FailureStorageTest
+{
+    @Rule
+    public final EphemeralFileSystemRule fs = new EphemeralFileSystemRule();
+    private FolderLayout folderLayout;
+    private final long indexId = 1;
+
+    @Before
+    public void before()
+    {
+        File rootDirectory = new File( "dir" );
+        fs.get().mkdirs( rootDirectory );
+        folderLayout = new FolderLayout( rootDirectory );
+        fs.get().mkdirs( folderLayout.getFolder( indexId ) );
```

Comment:
We should not do it here right? `FailureStorage` should be able to create required dirs itself during `reserveForIndex` call